engine/src/main/java/org/pentaho/di/base/AbstractBaseCommandExecutor.java

```diff
@@ -234,6 +237,31 @@ public boolean isEnabled( final String value ) {
     return YES.equalsIgnoreCase( value ) || Boolean.parseBoolean( value ); // both are NPE safe, both are case-insensitive
   }
 
+  /**
+   * Decodes the provided base64String into the specified filePath. Parent directories must already exist.
+   *
+   * @param base64String String BASE64 representation of a file
+   * @param filePath     String The path to which the base64String is to be decoded
+   * @param deleteOnExit boolean Delete created file on JVM exit if true
+   */
+  protected void decodeBase64StringToFile( String base64String, String filePath, boolean deleteOnExit ) {
```

Comment:
To make this method as reusable as possible, have it returning the `File` , instead of void.

That way you can even later one apply `File.getAbsolutePath()`, that you can use when creating that `zip:file://...` string.